### PR TITLE
Use nats instead of exocom

### PR DESCRIPTION
**Pull request by charlierudolph**
DO NOT MERGE - Proof of concept only

Requires working with master including the changes introduced in https://github.com/Originate/exosphere/pull/264

I created the `nats-connector` to reuse as much as I could of the existing services. You could easily write a different connecter if you desired. We are using the default port for client connections to nats.

screen shot 2017-06-06 at 5 22 14 pm

### CODE REVIEW

File: nats-connector/index.ls [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+
+  subscribeMapping: (messageCallbackMapping) ->
+    messageCallbackMapping |> obj-to-pairs |> each ([name, callback]) ~>
+      @nats.subscribe name, (requestDataStr, replyTo) ~>
+        requestData = JSON.parse requestDataStr
+        @_log 'received request for', name, requestData
+        callback requestData, reply: (replyData) ~>
+          @_log 'responding to', name, replyData
+          @nats.publish replyTo, JSON.stringify(replyData)
```
Comment (charlierudolph) [inline nats-connector/index.ls:21]: If you want the reply to also broadcast, (similar to what exocom does) this would need two publish calls.

**Review by kevgo — APPROVED** [review]
Excellent! Would be great to get this shipped and a new NPM module released today! I'll need this to talk to DBG tomorrow.

**Comment by kevgo** [comment]
I mean a NPM module for Exosphere that supports the `nats` dependency.

**Comment by charlierudolph** [comment]
Sounds good. We will work on getting a release out today

**Comment by hugoesthere** [comment]
@kevgo `0.22.1` is out! Just checkout this branch and `exo setup` + `exo run` should give you a running SpaceTweet application using Nats.

**Comment by kevgo** [comment]
@charlierudolph excellent work, thanks, I'll check it out!

**Comment by charlierudolph** [comment]
Give @hugobho credit for his help setting up the application dependencies, publishing exosphere, validating this branch worked with the latest release

**Comment by charlierudolph** [comment]
@kevgo do we want to keep this as a separate branch or what would you like to do with this?

**Comment by kevgo** [comment]
I am not sure. What would you propose? If it is easier, we can also make this a separate code base.

**Comment by charlierudolph** [comment]
Closing, but will keep as branch for now

**Pull request closed without merging**